File: app/src/main/java/com/wya/env/util/GestureFlingRightHelper.java
package com.wya.env.util;

import android.content.Context;
import android.view.GestureDetector;
import android.view.MotionEvent;

import com.wya.env.App;
import com.wya.utils.utils.ScreenUtil;

/**
 * @author: Caiqi Zen
 * @classname: GestureFlingRightHelper
 * @describe:
 */

public class GestureFlingRightHelper implements GestureDetector.OnGestureListener {
    private int minDistance = 180;
    private GestureRightCallBack mGestureCallBack = null;
    private static GestureFlingRightHelper mGfHelper;
    
    public GestureFlingRightHelper() {
    }
    
    public static GestureFlingRightHelper getInstance() {
        if (mGfHelper == null) {
            Class var0 = GestureFlingRightHelper.class;
            synchronized (GestureFlingRightHelper.class) {
                mGfHelper = new GestureFlingRightHelper();
            }
        }
        
        return mGfHelper;
    }
    
    public GestureDetector getGestureDetector(Context context, GestureRightCallBack callBack, int widthPixels) {
        minDistance = widthPixels / 3;
        mGestureCallBack = callBack;
        if (mGestureCallBack == null) {
            mGestureCallBack = new GestureRightCallBack() {
                @Override
                public boolean handleFlingRightAction() {
                    return false;
                }
            };
        }
        
        return new GestureDetector(context, this);
    }
    
    @Override
    public boolean onDown(MotionEvent e) {
        return false;
    }
    
    @Override
    public boolean onFling(MotionEvent e1, MotionEvent e2, float velocityX, float velocityY) {
        boolean result = false;
        if (e1 != null && e2 != null) {
            float xFling = e2.getX() - e1.getX();
            float yFling = e2.getY() - e1.getY();
            if (e1.getX() < (float) (ScreenUtil.getScreenWidth(App.getInstance()) / 5) && xFling > (float) minDistance && Math.abs(yFling) < xFling / 2.0F) {
                result = mGestureCallBack.handleFlingRightAction();
            }
        }
        
        return result;
    }
    
    @Override
    public void onLongPress(MotionEvent e) {
    }
    
    @Override
    public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float distanceY) {
        return false;
    }
    
    @Override
    public void onShowPress(MotionEvent e) {
    }
    
    @Override
    public boolean onSingleTapUp(MotionEvent e) {
        return false;
    }
    
    public interface GestureRightCallBack {
        /**
         * handleFlingRightAction
         *
         * @return
         */
        boolean handleFlingRightAction();
    }
}

File: app/src/main/java/com/wya/env/util/GestureFlingRightHelper.kt
package com.wya.env.util

import android.content.Context
import android.view.GestureDetector
import android.view.MotionEvent

import com.wya.env.App
import com.wya.utils.utils.ScreenUtil

/**
 * @author: Caiqi Zen
 * @classname: GestureFlingRightHelper
 * @describe:
 */

class GestureFlingRightHelper : GestureDetector.OnGestureListener {
    private var minDistance = 180
    private var mGestureCallBack: GestureRightCallBack? = null

    fun getGestureDetector(context: Context, callBack: GestureRightCallBack, widthPixels: Int): GestureDetector {
        minDistance = widthPixels / 3
        mGestureCallBack = callBack
        if (mGestureCallBack == null) {
            mGestureCallBack = object : GestureRightCallBack {
                override fun handleFlingRightAction(): Boolean {
                    return false
                }
            }
        }

        return GestureDetector(context, this)
    }

    override fun onDown(e: MotionEvent): Boolean {
        return false
    }

    override fun onFling(e1: MotionEvent?, e2: MotionEvent?, velocityX: Float, velocityY: Float): Boolean {
        var result = false
        if (e1 != null && e2 != null) {
            val xFling = e2.x - e1.x
            val yFling = e2.y - e1.y
            if (e1.x < (ScreenUtil.getScreenWidth(App.instance!!) / 5).toFloat() && xFling > minDistance.toFloat() && Math.abs(yFling) < xFling / 2.0f) {
                result = mGestureCallBack!!.handleFlingRightAction()
            }
        }

        return result
    }

    override fun onLongPress(e: MotionEvent) {}

    override fun onScroll(e1: MotionEvent, e2: MotionEvent, distanceX: Float, distanceY: Float): Boolean {
        return false
    }

    override fun onShowPress(e: MotionEvent) {}

    override fun onSingleTapUp(e: MotionEvent): Boolean {
        return false
    }

    interface GestureRightCallBack {
        /**
         * handleFlingRightAction
         *
         * @return
         */
        fun handleFlingRightAction(): Boolean
    }

    companion object {
        private var instances: GestureFlingRightHelper? = null

            get() {
                if (field == null) {
                    field = GestureFlingRightHelper()
                }

                return field
            }
        @Synchronized
        fun get():GestureFlingRightHelper{
            return instances!!
        }
    }
}
